### DISCUSSION TITLE
Exported type for `RemoteJWKSet`?

### DISCUSSION THREAD

**Discussion by ath88** [panva/jose#693] [Ideas]
We need to handle multiple URLs, as we can have multiple trusted IdPs. This means that we need to populate a list of the results from `createRemoveJWKSet`. As we define the list first, we need to specify the type on definition, as Typescript cannot infer it. The current return type from the function is[1]}:
``` ts
{
  (protectedHeader?: JWSHeaderParameters, token?: FlattenedJWSInput): Promise<KeyLikeType>
  /** @ignore */
  coolingDown: boolean
  /** @ignore */
  fresh: boolean
  /** @ignore */
  reloading: boolean
  /** @ignore */
  reload: () => Promise<void>
  /** @ignore */
  jwks: () => JSONWebKeySet | undefined
}
```

Defined here: https://github.com/panva/jose/blob/8d397cb1aadf99cc7981e668c5c41521723777ee/src/jwks/remote.ts#L333

The definition is very cumbersome to refer to, so could we have an exported type as `RemoteJWKSet`? Thoughts?

**Comment by panva** [comment]
Hi @ath88 

`ReturnType<typeof createRemoteJWKSet>` or `JWTVerifyGetKey`, which is already exported. `RemoteJWKSet` would be useless to you since you never touch that class directly.

**Reply by ath88** [reply]
You're right! Thanks for taking your time!
